src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -333,7 +341,14 @@ private void SetStartHelper(Delegate start, int maxStackSize)
         public bool IsBackground
         {
             get => IsBackgroundNative();
-            set => SetBackgroundNative(value);
+            set
+            {
+                SetBackgroundNative(value);
+                if (!value && !_mayNeedResetForThreadPool)
+                {
+                    _mayNeedResetForThreadPool = value;
```

Comment:
This looks like a no-op. Both `value` and `_mayNeedResetForThreadPool` will be fals when this branch is taken.

Should this be `_mayNeedResetForThreadPool = true` ?

Also, do we need a test to catch this?

Reply:
Nice catch, yea my intention was `= true`. Fixed and updated the test to verify each state reset separately.